### PR TITLE
New package: pvs-studio-7.07.37949.43_1

### PR DESCRIPTION
**Pull request by jbenden**
PVS-Studio is a static analysis tool for C/C++. It is commercial software.

Signed-off-by: Joseph Benden <joe@benden.us>

### CODE REVIEW

**Comment by Chocimier** [comment]
License forbids redistribution, so it could only be added as restricted (`repository=nonfree`, `restricted=yes`).  It means there is no binary packages in repo. One must use xbps-src to build package, then install from local repo, and there is no automatic updates.

**Comment by jbenden** [comment]
Thank you for catching the `restricted=yes` requirement.

Yes, it is less than ideal that redistributing is not possible; however, having a packaging script makes the installation possible without a bunch of manual (and repeated on upgrade) steps. Instead, a single build and install of the package is needed; using `xbps-src`.

**Comment by Chocimier** [comment]
Looks ok with one remark: license cannot be added into repo like that, it need to be downloaded to not become stale. What will be problematic as there is no copy in archive, and downloading webpage won't checksum.

File: srcpkgs/pvs-studio/template [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'pvs-studio'
+pkgname=pvs-studio
+version=7.05.35582.25
+revision=1
+archs="x86_64*"
```
Comment (Johnnynator) [inline srcpkgs/pvs-studio/template]:
```suggestion
archs="x86_64"
```
And don't set `broken`.

Comment (jbenden) [reply srcpkgs/pvs-studio/template]: Thanks for the review!

**Comment by jbenden** [comment]
I have made the following changes:

- I've upgraded the package version to the latest available, and modified the commit message to reflect the change.
- I've corrected the `arch` related items, per the code review.
- I have used a technique I found in the repository for downloading the license terms dynamically from the web at package creation.

Any comments, suggestions, and/or ideas are always welcomed,
-Joe

**Pull request closed without merging**